mackerel-plugin-aws-rekognition/lib/aws-rekognition.go

```diff
@@ -0,0 +1,167 @@
+package mpawswaf
+
+import (
+	"errors"
+	"flag"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	mp "github.com/mackerelio/go-mackerel-plugin"
+)
+
+var graphdef = map[string]mp.Graphs{
+	"rekognition.RequestCount": {
+		Label: "AWS Rekognition RequestCount",
+		Unit:  "integer",
+		Metrics: []mp.Metrics{
+			{Name: "SuccessfulRequestCount", Label: "SuccessfulRequestCount"},
+			{Name: "ThrottledCount", Label: "ThrottledCount"},
+			{Name: "ServerErrorCount", Label: "ServerErrorCount"},
+			{Name: "UserErrorCount", Label: "UserErrorCount"},
+		},
+	},
+	"rekognition.ResponseTime": {
+		Label: "AWS Rekognition ResponseTime",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "ResponseTime", Label: "ResponseTime"},
+		},
+	},
+	"rekognition.DetectedCount": {
+		Label: "AWS Rekognition DetectedCount",
+		Unit:  "integer",
+		Metrics: []mp.Metrics{
+			{Name: "DetectedFaceCount", Label: "DetectedFaceCount"},
+			{Name: "DetectedLabelCount", Label: "DetectedLabelCount"},
+		},
+	},
+}
+
+// RekognitionPlugin mackerel plugin for aws rekognition
+type RekognitionPlugin struct {
+	Region          string
+	AccessKeyID     string
+	SecretAccessKey string
+	CloudWatch      *cloudwatch.CloudWatch
+}
+
+func (p *RekognitionPlugin) prepare() error {
+	sess, err := session.NewSession()
+	if err != nil {
+		return err
+	}
+
+	config := aws.NewConfig()
+	if p.AccessKeyID != "" && p.SecretAccessKey != "" {
+		config = config.WithCredentials(credentials.NewStaticCredentials(p.AccessKeyID, p.SecretAccessKey, ""))
+	}
+	if p.Region != "" {
+		config = config.WithRegion(p.Region)
+	}
+
+	p.CloudWatch = cloudwatch.New(sess, config)
+
+	return nil
+}
+
+func (p RekognitionPlugin) getLastPoint(metricName string) (float64, error) {
+	now := time.Now()
+
+	response, err := p.CloudWatch.GetMetricStatistics(&cloudwatch.GetMetricStatisticsInput{
```

Comment:
According to [CloudWatch Reference](http://docs.aws.amazon.com/rekognition/latest/dg/cloudwatch-metricsdim.html), Rekognition metrics can take dimensions "Operations" optionally. Is it useful if this plugin supports operation dimensions? (like recording RequestCount metrics for each operation.)

> To retrieve operation-specific metrics, use the Rekognition namespace and provide an operation dimension.